solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/ModelQuery.java

```diff
@@ -259,26 +336,33 @@ public Explanation explain(LeafReaderContext context, int doc)
 
     @Override
     public void extractTerms(Set<Term> terms) {
-      for (final FeatureWeight feature : allFeatureWeights) {
+      for (final FeatureWeight feature : extractedFeatureWeights) {
         feature.extractTerms(terms);
       }
     }
 
     protected void reset() {
-      for (int i = 0, len = allFeaturesUsed.length; i < len; i++) {
-        allFeaturesUsed[i] = false;
-        allFeatureValues[i] = allFeatureWeights[i].getDefaultValue();
-      }
+       for (int i = 0; i < extractedFeatureWeights.length;++i){
+          String featName = extractedFeatureWeights[i].getName();
+          int featId = extractedFeatureWeights[i].getId();
+          float value = extractedFeatureWeights[i].getDefaultValue();
+          if (featuresInfo[featId] == null){
+            featuresInfo[featId] = new FeatureInfo(featName,value,false);
+          }
+          else{
+            featuresInfo[featId].setAll(featName, value, false);
```

Comment:
This line is called for all features for all docs scored, so it seems like it's kinda nested.  Do we need to do string + value assignment every time, of can we just do a boolean used=false on reset?  You would have to change the constructor to not call reset() anymore, but instead create all the FeatureInfo objects once.